src/foam/dao/SinkJava.js

```diff
@@ -166,7 +166,7 @@ foam.CLASS({
     {
       name: 'eof',
       javaCode: 'if ( getArray() == null ) setArray(new java.util.ArrayList());\n'
-                + 'java.util.Collections.sort(getArray());\n'
+                + 'java.util.Collections.sort(getArray(), comparator_);\n'
```

Comment:
Use getComparator() instead.  It's preferable to access properties through the getFoo() interface, even if they are local properties so we don't skip any additional getter logic that FOAM properties may provide.